test/avocado/selenium-login.py

```diff
@@ -32,101 +32,141 @@
 import inspect
 import selenium.webdriver
 from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
 import os
+import sys
 import re
 import time
+machine_test_dir = os.path.dirname(
+    os.path.abspath(inspect.getfile(inspect.currentframe())))
+sys.path.append(machine_test_dir)
+import libdisc
 
 user = "test"
 passwd = "superhardpasswordtest5554"
+javascript_clicks=True
 
+present = EC.presence_of_element_located
+visible = EC.visibility_of_element_located
+clickable = EC.element_to_be_clickable
+invisible = EC.invisibility_of_element_located
+frame = EC.frame_to_be_available_and_switch_to_it
 
-class BasicTestSuite(Test):
 
-    def __init__(self, *args, **kwargs):
-        super(BasicTestSuite, self).__init__(*args, **kwargs)
+class BasicTestSuite(Test):
 
     def setUp(self):
         if not (os.environ.has_key("HUB") or os.environ.has_key("BROWSER")):
             self.driver = selenium.webdriver.Firefox()
             guest_machine = 'localhost'
         else:
-            selenium_hub = os.environ["HUB"] if os.environ.has_key("HUB") else "localhost"
-            browser = os.environ["BROWSER"] if os.environ.has_key("BROWSER") else "firefox"
+            selenium_hub = os.environ["HUB"] if os.environ.has_key(
+                "HUB") else "localhost"
+            browser = os.environ["BROWSER"] if os.environ.has_key(
+                "BROWSER") else "firefox"
             guest_machine = os.environ["GUEST"]
             self.driver = selenium.webdriver.Remote(
                 command_executor='http://%s:4444/wd/hub' % selenium_hub, desired_capabilities={'browserName': browser})
 
-        self.driver.set_window_size(1024, 768)
+        self.driver.set_window_size(1400, 1200)
         self.driver.set_page_load_timeout(90)
-        self.driver.implicitly_wait(10)
-        self.default_try = 10
-        self.default_sleep = 1
+        self.default_try = 40
+        self.default_explicit_wait = 1
         self.driver.get('http://%s:9090' % guest_machine)
+        self.error=True
 
     def tearDown(self):
-        pass
-        self.driver.close()
-        self.driver.quit()
+        if self.error:
+            screenshot_file=""
+            try:
+                screenshot_file="snapshot-teardown-%s.png" %  str(time.clock())
```

Comment:
Because in tearDown phase I'm unable to find what test was running, so that I need some timestamp in case there will me more screenshots. This screenshosts were generally caused by click() problem.